src/id/doujindesu/src/eu/kanade/tachiyomi/extension/id/doujindesu/DoujinDesu.kt

```diff
@@ -0,0 +1,320 @@
+                    url.addQueryParameter("type", category.key)
+                }
+                is OrderBy -> {
+                    val order = filter.values[filter.state]
+                    url.addQueryParameter("order", order.key)
+                }
+                is GenreList -> {
+                    val genre = filter.values[filter.state]
+                    url.addQueryParameter("genre", genre.key)
+                }
+                is StatusList -> {
+                    val status = filter.values[filter.state]
+                    url.addQueryParameter("status", status.key)
+                }
+            }
+        }
+        return GET(url.toString(), headers)
+    }
+
+    override fun searchMangaFromElement(element: Element): SManga = basicInformationFromElement(element)
+
+    override fun getFilterList() = FilterList(
+        CategoryNames(categoryNames),
+        OrderBy(orderBy),
+        GenreList(genreList),
+        StatusList(statusList)
+    )
+
+    // Detail Parse
+
+    override fun mangaDetailsParse(document: Document): SManga {
+        val manga = SManga.create()
+        manga.description = when {
+            document.select("div.infox > div.entry-content.entry-content-single > p").isEmpty() -> "No description specified"
+            else -> document.select("div.infox > div.entry-content.entry-content-single > p").first().text()
+        }
+        manga.author = document.select("div.infox > div.spe > span:nth-child(5)").text()
+        manga.genre = document.select("div.genre-info > a[itemprop=genre]").joinToString { it.text() }
+        manga.status = parseStatus(document.select("div.infox > div.spe > span:nth-child(1)").text())
+        manga.thumbnail_url = document.select("div.thumb > img").attr("src")
+        manga.artist = document.select("div.infox > div.spe > span:nth-child(6)").text()
+
+        return manga
+    }
+
+    // Chapter Stuff
+
+    override fun chapterFromElement(element: Element): SChapter {
+        val chapter = SChapter.create()
+        val number = getNumberFromString(element.select("div.epsright > span > a > chapter").text())
+        chapter.chapter_number = when {
+            (number.isNotEmpty()) -> number.toFloat()
+            else -> 1F
+        }
+        chapter.date_upload = reconstructDate(element.select("div.epsleft > span.date").text())
+        chapter.name = element.select("div.epsleft > span.lchx > a").text()
+        chapter.setUrlWithoutDomain(element.select("div.epsleft > span.lchx > a").attr("href"))
+
+        return chapter
+    }
+
+    override fun chapterListSelector(): String = "#chapter_list li"
+
+    // More parser stuff
+    override fun imageUrlParse(document: Document): String = throw UnsupportedOperationException("Not Used")
+
+    override fun pageListParse(document: Document): List<Page> {
+        return document.select("div.reader-area > img").map { element ->
+            getImage(element)
+        }
+    }
```

Comment:
```suggestion
    override fun pageListParse(document: Document): List<Page> {
        return document.select("div.reader-area > img").mapIndexed { i, element ->
            Page(i, "", element.attr("src"))
        }
    }
```

You can then remove the `getImage` and the `getNumberFromString` methods.